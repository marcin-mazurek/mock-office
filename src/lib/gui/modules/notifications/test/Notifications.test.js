import React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

jest.doMock('cuid', () => () => 'id');
const { Notifications, createNotification } = require('../lib');

describe('NotificationsList', () => {
  describe('snapshots', () => {
    test('without notifications', () => {
      const props = {
        notifications: [],
        onNotificationClick: () => {},
        onNewNotificationDisplayed: () => {}
      };
      const wrapper = shallow(<Notifications {...props} />);
      expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    test('with notification', () => {
      const notification = createNotification({ text: 'message' });
      const props = {
        notifications: [notification],
        onNotificationClick: () => {},
        onNewNotificationDisplayed: () => {}
      };
      const wrapper = shallow(<Notifications {...props} />);
      expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
  });
});
