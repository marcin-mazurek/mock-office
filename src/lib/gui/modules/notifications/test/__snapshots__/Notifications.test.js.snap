// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsList snapshots with notification 1`] = `
<ul
  className="notifications"
>
  <li
    className="notification__item"
  >
    <Notification
      id="id"
      onNotificationClick={[Function]}
      text="message"
      type="info"
    />
  </li>
</ul>
`;

exports[`NotificationsList snapshots without notifications 1`] = `
<ul
  className="notifications"
/>
`;
